lib/user.js

```diff
@@ -52,6 +54,45 @@ class UserModel extends BaseModel {
     unsealToken(sealed, callback) {
         return iron.unseal(sealed, this.password, iron.defaults, callback);
     }
+
+    /**
+     * Get permissions on a specific repo
+     * @param  {String}  config.username        Username
+     * @param  {String}  config.scmUrl          The scmUrl of the repository
+     * @return {Function} callback              fn(err, permissions) where permissions is an object
+     *                                          that contains the permissions for [admin, push, pull]
+     *                                          Example: {admin: false, push: true, pull: true}
+     */
+    getPermissions(config, callback) {
+        const id = this.generateId({ username: config.username });
+
+        async.waterfall([
+            (next) => {
+                this.get(id, next);
+            },
+            (user, next) => {
+                this.unsealToken(user.token, next);
+            },
+            (unsealed, next) => {
+                const matched = config.scmUrl.match(/^git@([^:]+):([^\/]+)\/(.+?)\.git(\/.+)?$/);
+
+                github.authenticate({
+                    type: 'oauth',
+                    token: unsealed
+                });
+                github.repos.get({
+                    user: matched[2],
```

Comment:
Nope. this would check again the user with the githubtoken = `unsealed` against this particular repo.